### PR TITLE
Fix running workflow tests when there are multiple tests

### PR DESCRIPTION
**Pull request by mvdbeek**
and also re-uses the galaxy instance for additional test.

Fixes https://github.com/galaxyproject/iwc/pull/274#issuecomment-1807868948

### CODE REVIEW

**Comment by lldelisle** [comment]
Failed lint is in planemo/engine/interface.py:
```
              output_collectors.append(
-                lambda run_response, test_case=test_case: test_results.append(test_case.structured_test_data(run_response))
+                lambda run_response, test_case=test_case: test_results.append(
+                    test_case.structured_test_data(run_response)
+                )
             )
  ```

File: tests/test_cmd_test.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def test_data_manager_docker_mount(self):
     @skip_if_environ("PLANEMO_SKIP_GALAXY_TESTS")
     def test_workflow_test_simple_yaml(self):
         """Test testing a simple YAML workflow with Galaxy."""
-        with self._isolate():
+        with self._isolate() as test_dir:
             random_lines = os.path.join(PROJECT_TEMPLATES_DIR, "demo", "randomlines.xml")
```
Comment (nsoranzo) [inline tests/test_cmd_test.py:93]: Not related, but why is the `random_lines1` tool added as `--extra_tools` below (and in another test), even if the tool is not used in the `wf1.gxwf.yml` workflow?

Comment (mvdbeek) [reply tests/test_cmd_test.py:93]: maybe to check that multiple `--extra_tools` work ?

**Comment by mvdbeek** [comment]
Test error seems unrelated, quay.io timing out

**Pull request merged**